src/trimerge-array.ts

```diff
@@ -17,73 +16,45 @@ export function trimergeArrayCreator(
     right: any[],
     path: Path,
     mergeFn: MergeFn,
-  ): JSONValue[] | typeof CannotMerge => {
+  ): any[] | typeof CannotMerge => {
     if (jsSameType(orig, left, right) !== 'array') {
       return CannotMerge;
     }
-    const origMap: Record<string, any> = {};
-    const leftMap: Record<string, any> = {};
-    const rightMap: Record<string, any> = {};
-    const origKeys = orig.map((item, index): string => {
-      const key = getArrayItemKey(item, index, path);
-      if (key in origMap) {
-        throw new Error(`Duplicate array key '${key}' at /${path}`);
-      }
-      origMap[key] = item;
-      return key;
-    });
-    const leftKeys = left.map((item, index): string => {
-      const key = getArrayItemKey(item, index, path);
-      if (key in leftMap) {
-        throw new Error(`Duplicate array key '${key}' at /${path}`);
-      }
-      leftMap[key] = item;
-      return key;
-    });
-    const rightKeys = right.map((item, index): string => {
-      const key = getArrayItemKey(item, index, path);
-      if (key in rightMap) {
-        throw new Error(`Duplicate array key '${key}' at /${path}`);
-      }
-      rightMap[key] = item;
-      return key;
-    });
 
-    const mergedArray: JSONValue[] = [];
-    let leftSame = true;
-    let rightSame = true;
-    diff3Keys(
-      origKeys,
-      leftKeys,
-      rightKeys,
-      (key) => {
-        const merged = mergeFn(
-          origMap[key],
-          leftMap[key],
-          rightMap[key],
-          [...path, key],
-          mergeFn,
-        );
-        // Compare merge result with same array index in left
-        if (leftSame && merged !== left[mergedArray.length]) {
-          leftSame = false;
+    function arrayToMap(array: any[]): Map<string, any> {
```

Comment:
Might be cleaner to define this in the global scope and pass `getArrayItemKey` and `path` in as arguments. 🤷‍♂

Reply:
tried it out, ends up being more verbose